### ISSUE TITLE
`where_or` for Query and QueryProxy

### ISSUE THREAD

**Issue by cheerfulstoic** [neo4jrb/neo4j#947]

**Comment by subvertallchris** [comment]
:+1

**Comment by cheerfulstoic** [comment]
Took off `low-hanging-fruit` tag.  Trying to do this made me think that the `Query` / `QueryClauses` stuff should perhaps be refactored...

;)

**Comment by dapicester** [comment]
Does it mean that currently is not possible to use `OR` conditions when querying `ActiveModel` classes?

**Comment by cheerfulstoic** [comment]
It's certainly possible, you just need to give a string.  Something like:

```ruby
person.as(:p).where("p.age = 20 OR p.age = 30")
```

**Comment by dapicester** [comment]
Of course with a string can write Cypher.

I am trying to write a Neo4j adapter for [cancancan](https://github.com/CanCanCommunity/cancancan) (used by [rails_admin](https://github.com/sferik/rails_admin)) so I need to build the query dynamically and I am using ActiveModel methods. Maybe I will try with cypher string ...

**Comment by cheerfulstoic** [comment]
First I'm a bit confused when you say `ActiveModel`.  Do you mean `ActiveNode` (which uses `ActiveModel` to access Neo4j nodes)?

If you haven't seen it already, you should look into the `Query` class from `neo4j-core` which allows you to build cypher queries via Ruby methods.  You can get a fresh query like `Neo4j::Session.current.query` or you can build one off of `ActiveNode` like `person.association.as_query(:association)` or `Person.association.as_query(:association)`, etc...

Also, I've worked on and off to get a neo4j version of rails_admin working here:
https://github.com/cheerfulstoic/rails_admin

It works more or less, I've been trying to get the specs to all pass now.

Great to hear that you're helping to support Neo4j with cancancan!

**Comment by dapicester** [comment]
Yes, sorry I mean `ActiveNode`. 

I tried your fork of rails_admin you mentioned on sferik/rails_admin#1377 and more or less it works for me but I have to disable authorizations (basically comment out the `config.authorize_with :cancan, RailsAdmin::Ability` line). That's why I am trying to add Neo4j support in `cancancan`.

Of course if I can get something working I will submit PR and let you know.

**Comment by dapicester** [comment]
I am reading the `Query` and `QueryClauses`, but so far it seems to me that there is no support for the OR clause unless writing Cyper.

What I mean i something similar to [this Rails PR](https://github.com/rails/rails/pull/16052) where one can write
```ruby
Article.where(published: true).or(Article.where(secret: true))
```
or using the `where_or` syntax
```ruby
Article.where(published: true).where_or(secret: true)
```
and get something like
```cypher
MATCH (n:Article) WHERE n.published = true OR n.secret = false
```

Do you think can it be (easily) done in the `neo4j` gem?

**Comment by cheerfulstoic** [comment]
I don't think it will be easily done, unfortunately.  I've generated Cyhper `OR` statements programmatically before, though, so I would suggest seeing if you can do that.
